contextual_lenses/pfam_family_classification/pfam_one_hot_lenses_parallel.py

```diff
@@ -0,0 +1,223 @@
+"""Contextual lenses on Pfam family classification dataset (first 100 families)."""
+
+
+"""Parse command line arguments."""
+from parser import parse_args
+tpu_name, save_dir, restore_dir = parse_args()
+
+
+"""Connect TPU to VM instance."""
+from cloud_utils.tpu_init import connect_tpu
+connect_tpu(tpu_name=tpu_name)
+
+
+import functools
+import itertools
+import os
+import time
+
+import flax
+from flax import nn
+
+import jax
+import jax.numpy as jnp
+
+import tensorflow as tf
+
+import numpy as np
+
+import pandas as pd
+
+import matplotlib.pyplot as plt
+
+import scipy.stats
+
+import sklearn.linear_model
+import sklearn.metrics
+
+import seaborn as sns
+
+from contextual_lenses import mean_pool, max_pool, \
+linear_max_pool, linear_mean_pool, gated_conv 
+
+from train_utils import create_optimizer, p_train, \
+create_representation_model
+
+from encoders import one_hot_encoder, cnn_one_hot_encoder, \
+one_hot_pos_emb_encoder, cnn_one_hot_pos_emb_encoder
+
+from loss_fns import cross_entropy_loss
+
+
+"""Data preprocessing
+Code source: https://www.kaggle.com/drewbryant/starter-pfam-seed-random-split."""
+data_partitions_dirpath = 'random_split/random_split/'
+print('Available dataset partitions: ', os.listdir(data_partitions_dirpath))
+
+def read_all_shards(partition='dev', data_dir=data_partitions_dirpath):
+    shards = []
+    for fn in os.listdir(os.path.join(data_dir, partition)):
+        with open(os.path.join(data_dir, partition, fn)) as f:
+            shards.append(pd.read_csv(f, index_col=None))
+    return pd.concat(shards)
+
+test_df = read_all_shards('test')
+dev_df = read_all_shards('dev')
+train_df = read_all_shards('train')
+
+partitions = {'test': test_df, 'dev': dev_df, 'train': train_df}
+for name, df in partitions.items():
+    print('Dataset partition "%s" has %d sequences' % (name, len(df)))
+
+def mod_family_accession(family_accession):
+  return family_accession[:family_accession.index('.')]
+
+train_df['mod_family_accession'] = train_df.family_accession.apply(lambda x: mod_family_accession(x))
+test_df['mod_family_accession'] = test_df.family_accession.apply(lambda x: mod_family_accession(x))
+dev_df['mod_family_accession'] = dev_df.family_accession.apply(lambda x: mod_family_accession(x))
+
+kept_family_accessions = []
+for i in range(1, 101):
+  kept_family_accessions.append('PF00' + '0'*(3-len(str(i))) + str(i))
+
+def keep_family_accessions(df, kept_family_accessions):
```

Comment:
you can replace this  function with df[df.mod_family_accession.isin(kept_family_accessions)]